docs/en/development-and-contribution/development-guide.md

```diff
@@ -0,0 +1,333 @@
+# Plugin Development Guide
+
+This documentation introduces how developers can create a plugin.
+
+All plugins must follow these steps:
+
+1. **Create a new plugin module**: Create a new project in the specified directory and import the plugin API module.
+2. **Define the enhancement object**: Define the description for the plugin.
+3. **Invoke the plugin API**: Call the API provided by the core to complete the core invocation.
+4. **Import the plugin module**: Import the plugin into the management module for users to use.
+
+## Create a new plugin module
+
+The plugin must create a new module, which is currently stored in the project's [plugins directory](../../../plugins).
+
+Plugins can import the following two modules:
+
+1. **Agent core**: This module provides all the dependencies needed for the plugin, including the plugin API, enhancement declaration objects, etc.
+Agent core plugin should be `github.com/apache/skywalking-go/plugins/core` and replaced by the relative location.
+2. **Framework to be enhanced**: Import the framework you wish to enhance.
+
+Note: Plugins should not import and use other unrelated modules, as this may cause compilation issues for users. If certain tools are needed, they should be provided by the agent core.
```

Comment:
```suggestion
Note: Plugins should **NOT** import and use any other modules, as this may cause compilation issues for users. If certain tools are needed, they should be provided by the agent core.
```